src/Providers/ScoutApmServiceProvider.php

```diff
@@ -1,47 +1,56 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Scoutapm\Laravel\Providers;
 
-use Illuminate\Contracts\Http\Kernel;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Contracts\Foundation\Application;
 use Illuminate\Contracts\View\Engine;
+use Illuminate\Contracts\View\View;
+use Illuminate\Database\Connection;
 use Illuminate\Database\Events\QueryExecuted;
-use Illuminate\Support\Facades\DB;
+use Illuminate\Foundation\Http\Kernel;
 use Illuminate\Support\ServiceProvider;
-
 use Illuminate\View\Engines\EngineResolver;
-use Scoutapm\Agent;
+use Illuminate\View\Factory as ViewFactory;
 use Psr\Log\LoggerInterface;
-
+use Scoutapm\Agent;
 use Scoutapm\Config;
-use Scoutapm\Laravel\Commands\DownloadCoreAgent;
-use Scoutapm\Laravel\Events\ScoutViewEngineDecorator;
+use Scoutapm\Laravel\Database\QueryListener;
 use Scoutapm\Laravel\Middleware\ActionInstrument;
+use Scoutapm\Laravel\Middleware\IgnoredEndpoints;
 use Scoutapm\Laravel\Middleware\MiddlewareInstrument;
 use Scoutapm\Laravel\Middleware\SendRequestToScout;
-use Scoutapm\Laravel\Middleware\IgnoredEndpoints;
-
+use Scoutapm\Laravel\View\Engine\ScoutViewEngineDecorator;
+use Scoutapm\ScoutApmAgent;
 
-class ScoutApmServiceProvider extends ServiceProvider
+final class ScoutApmServiceProvider extends ServiceProvider
 {
-    public function register()
-    {
-        // No events currently mapped. May be needed in the future.
-        // $this->app->register(EventServiceProvider::class);
+    private const SCOUTAPM_ALIAS_SERVICE_KEY = 'scoutapm';
+
+    private const VIEW_ENGINES_TO_WRAP = ['file', 'php', 'blade'];
 
-        $this->app->singleton(Agent::class, function ($app) {
+    /** @throws BindingResolutionException */
+    public function register() : void
+    {
+        $this->app->singleton(ScoutApmAgent::class, static function (Application $app) {
             return Agent::fromConfig(
                 new Config(),
-                $this->app->make('log')
+                $app->make('log')
             );
         });
 
-        $this->app->alias(Agent::class, 'scoutapm');
+        // @todo note - these aliases are only for backwards compatibility; but since we're early on, we could just remove?
```

Comment:
Thoughts on this appreciated. Do we need different keys aliased, or could we just enforce to use the `\Scoutapm\ScoutApmAgent::class` interface reference?

Reply:
Yeah, we can remove and update. We'll need to tweak the docs.